go/systests/team_tx_test.go

```diff
@@ -0,0 +1,307 @@
+package systests
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/teams"
+)
+
+func TestTeamTx1(t *testing.T) {
+	tt := newTeamTester(t)
+	defer tt.cleanup()
+
+	ann := makeUserStandalone(t, "ann", standaloneUserArgs{
+		disableGregor:            true,
+		suppressTeamChatAnnounce: true,
+	})
+	tt.users = append(tt.users, ann)
+	t.Logf("Signed up ann (%s)", ann.username)
+
+	bob := tt.addPuklessUser("bob")
+	t.Logf("Signed up PUK-less user bob (%s)", bob.username)
+
+	tracy := tt.addUser("trc")
+	t.Logf("Signed up PUK-ful user trc (%s)", tracy.username)
+
+	team := ann.createTeam()
+	t.Logf("Team created (%s)", team)
+
+	// TRANSACTION 1 - add bob (keybase-type invite) and tracy (crypto member)
+
+	teamObj := ann.loadTeam(team, true /* admin */)
+
+	tx := teams.CreateAddMemberTx(teamObj)
+	tx.AddMemberByUsername(context.Background(), bob.username, keybase1.TeamRole_WRITER)
+	tx.AddMemberByUsername(context.Background(), tracy.username, keybase1.TeamRole_READER)
+
+	err := tx.Post(context.Background())
+	require.NoError(t, err)
+
+	teamObj = ann.loadTeam(team, true /* admin */)
+	require.Equal(t, 1, teamObj.NumActiveInvites())
+	invites := teamObj.GetActiveAndObsoleteInvites()
+	require.Equal(t, 1, len(invites))
+	for _, invite := range teamObj.GetActiveAndObsoleteInvites() {
+		uv, err := invite.KeybaseUserVersion()
+		require.NoError(t, err)
+		require.EqualValues(t, bob.userVersion(), uv)
+	}
+
+	members, err := teamObj.Members()
+	require.NoError(t, err)
+	require.Equal(t, 1, len(members.Owners))
+	require.Equal(t, 0, len(members.Admins))
+	require.Equal(t, 0, len(members.Writers))
+	require.Equal(t, 1, len(members.Readers))
+	require.EqualValues(t, tracy.userVersion(), members.Readers[0])
+
+	// TRANSACTION 2 - bob gets puk, add bob but not through SBS - we
+	// expect the invite to be sweeped away by this transaction.
+
+	bob.perUserKeyUpgrade()
+
+	teamObj = ann.loadTeam(team, true /* admin */)
+	tx = teams.CreateAddMemberTx(teamObj)
+	tx.AddMemberByUsername(context.Background(), bob.username, keybase1.TeamRole_WRITER)
+
+	err = tx.Post(context.Background())
+	require.NoError(t, err)
+
+	teamObj = ann.loadTeam(team, true /* admin */)
+	members, err = teamObj.Members()
+	require.NoError(t, err)
+	require.Equal(t, 1, len(members.Owners))
+	require.Equal(t, 0, len(members.Admins))
+	require.Equal(t, 1, len(members.Readers))
+	require.Equal(t, 1, len(members.Writers))
+	require.EqualValues(t, bob.userVersion(), members.Writers[0])
+	require.Equal(t, 0, len(teamObj.GetActiveAndObsoleteInvites()))
```

Comment:
It's when team player decides that invite is not active because a change_membership link for that user happened afterwards. So we treat these as "obsolete" invites - they were not canceled or completed but client will not see them e.g. in SBS handler. We didn't want to re-use term "active/inactive".

Reply:
Hmm, but isn't this `mooted`?

Reply:
I guess it's like `moot`ing but client (player) side. It was named "obsolete" after some discussion offline/online https://github.com/keybase/client/pull/10171